exec/java-exec/src/test/java/org/apache/drill/exec/expr/fn/impl/TestStringFunctions.java

```diff
@@ -114,6 +114,19 @@ public void testRegexpMatches() throws Exception {
   }
 
   @Test
+  public void testRegexpReplace() throws Exception {
+    testBuilder()
+        .sqlQuery("select regexp_replace(a, 'a|c', 'x') res1, regexp_replace(b, 'd', 'zzz') res2 " +
+                  "from (values('abc', 'bcd'), ('bcd', 'abc')) as t(a,b)")
+        .unOrdered()
+        .baselineColumns("res1", "res2")
+        .baselineValues("xbx", "bczzz")
+        .baselineValues("bxd", "abc")
+        .build()
+        .run();
+  }
+
```

Comment:
Can you add at least one case for the non-ASCII case, since one of the issue this PR tries to address is the incorrect result for regex function over non-ASCII input?  

A new unit test case will help verify this PR fix the problem, and make sure any new change will not regress this functionality.